### ISSUE TITLE
Why tensorflow don't support tf.float64 on some ops?

### ISSUE THREAD

**Issue by gauss-clb** [tensorflow/tensorflow#15039]
Reproducing experiment of some paper using tensorflow may don't have good performance. I have tried my best to keep the structure and hyper-parameters unchanged. Does anyone encounter similar problem? Is it the code wrong?  I wonder that `caffe` can use `float64` but tensorflow only can use `float32`,  so there are more inaccuracy on `gradients`, why tf don't support `tf.float64`?
